Step3_correct_background/calculate_bias.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+import xarray as xr
+#######################################################
+# This script takes the preprocessed daily data 
+# from tropomi/GC simulation and calculates the bias 
+# between the two using the latitudinal averages and 
+# spatial/temporal smoothing. 
+# This script is based off of Lu Shen's 
+# methods/ R script.
+#######################################################
+
+# settings to adjust default smoothing windows
+# we use a +/- 15 day time window and 3x3
+# smoothing for lat/lon
+smoothing_lat_window = 3
+smoothing_lon_window = 3
+smoothing_time_window = 30
+
+# access the preprocessed CH4 data
+filepath = "Daily_CH4.nc"
+daily_CH4 = xr.open_dataset(filepath)
+TROPOMI_lon = daily_CH4["lon"]
+TROPOMI_lat = daily_CH4["lat"]
+GC_CH4 = daily_CH4["GC"]
+OMI_CH4 = daily_CH4["CH4"]
+date = daily_CH4["date"]
+daily_CH4.close()
+
+
+# smooth out 3d dataarray over time, lat, lon using desired windows
+def smooth_3D_da(
+    da,
+    t_window=smoothing_time_window,
+    lat_window=smoothing_lat_window,
+    lon_window=smoothing_lon_window,
+):
+    return da.rolling(
+        time=t_window,
+        lat=lat_window,
+        lon=lon_window,
+        min_periods=1,
+        center=True,
+    ).mean(skipna=True)
+
+
+# smooth the background GC and TROPOMI data
+GC_bkgd = smooth_3D_da(GC_CH4)
+TROPOMI_bkgd = smooth_3D_da(OMI_CH4)
+
+# calculate bias between GC background CH4 and
+# TROPOMI observational background CH4
+bias_4x5 = GC_bkgd - TROPOMI_bkgd
+
+# build a smoothed dataset to fill in nan values of raw bias
+# start by smoothing the bias
+bias_avg_base = smooth_3D_da(bias_4x5)
```

Comment:
Why do we smooth `bias_4x5`? It's the difference between two already smoothed fields. Just checking.

Reply:
This one is an error -- removed!